internal/services/recoveryservices/site_recovery_vmware_replicated_vm_resource.go

```diff
@@ -0,0 +1,1204 @@
+			}
+
+			if model.DefaultLogStorageAccountId != "" {
+				diskDefaultValueSet = true
+				diskDefaultValue.LogStorageAccountId = model.DefaultLogStorageAccountId
+			}
+
+			if model.DefaultTargetDiskEncryptionSetId != "" {
+				diskDefaultValueSet = true
+				diskDefaultValue.DiskEncryptionSetId = &model.DefaultTargetDiskEncryptionSetId
+			}
+
+			if diskDefaultValueSet {
+				providerSpecificDetail.DisksDefault = &diskDefaultValue
+			}
+
+			if model.TargetAvailabilitySetId != "" {
+				providerSpecificDetail.TargetAvailabilitySetId = pointer.To(model.TargetAvailabilitySetId)
+			}
+
+			if model.TargetZone != "" {
+				providerSpecificDetail.TargetAvailabilityZone = pointer.To(model.TargetZone)
+			}
+
+			if model.TargetBootDiagnosticsStorageAccountId != "" {
+				providerSpecificDetail.TargetBootDiagnosticsStorageAccountId = pointer.To(model.TargetBootDiagnosticsStorageAccountId)
+			}
+
+			if model.MultiVmGroupName != "" {
+				providerSpecificDetail.MultiVMGroupName = &model.MultiVmGroupName
+			}
+
+			if model.TargetProximityPlacementGroupId != "" {
+				providerSpecificDetail.TargetProximityPlacementGroupId = &model.TargetProximityPlacementGroupId
+			}
+
+			if model.ApplianceName != "" {
+				providerSpecificDetail.ProcessServerId = processServerId
+			}
+
+			// split test network and subnet into isolated parameters
+			if model.TargetNetworkId != "" {
+				providerSpecificDetail.TargetNetworkId = &model.TargetNetworkId
+			}
+
+			if model.TestNetworkId != "" {
+				providerSpecificDetail.TestNetworkId = &model.TestNetworkId
+			}
+
+			diskToIncludeOutput := make([]replicationprotecteditems.InMageRcmDiskInput, 0)
+			for _, diskRaw := range model.DiskToInclude {
+				diskToIncludeOutput = append(diskToIncludeOutput, replicationprotecteditems.InMageRcmDiskInput{
+					DiskEncryptionSetId: &diskRaw.TargetDiskEncryptionSetId,
+					DiskId:              diskRaw.DiskId,
+					DiskType:            replicationprotecteditems.DiskAccountType(diskRaw.TargetDiskType),
+					LogStorageAccountId: diskRaw.LogStorageAccountId,
+				})
+			}
+
+			if len(diskToIncludeOutput) > 0 {
+				providerSpecificDetail.DisksToInclude = &diskToIncludeOutput
+			}
+
+			parameters := replicationprotecteditems.EnableProtectionInput{
+				Properties: &replicationprotecteditems.EnableProtectionInputProperties{
+					PolicyId:                &model.RecoveryReplicationPolicyId,
+					ProviderSpecificDetails: providerSpecificDetail,
+				},
+			}
+
+			poller, err := client.Create(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %q: %+v", id, err)
+			}
+			metadata.SetID(id) // once the PUT request returned successfully, an item has been created, even if it may fail in the poll process.
+
+			err = poller.Poller.PollUntilDone()
+			if err != nil {
+				return fmt.Errorf("polling %q: %+v", id, err)
+			}
+
+			err = resourceSiteRecoveryReplicatedVmVMWareClassicUpdateInternal(ctx, metadata)
```

Comment:
Why do we need an update after creation? As those `HasChanged` should all return false in this case.